### PR TITLE
Show a warning when a withdrawal address does not match the sender address

### PR DESCRIPTION
**Pull request by pienkowb**

### CODE REVIEW

File: src/handle_finalize.c [hunk 1]
```diff
@@ -1,10 +1,30 @@
...
 void handle_finalize(void *parameters) {
     ethPluginFinalize_t *msg = (ethPluginFinalize_t *) parameters;
+    context_t *context = (context_t *) msg->pluginContext;
 
     msg->uiType = ETH_UI_TYPE_GENERIC;
-
     msg->numScreens = 1;
+
+    // Copy the withdrawal address (if present)
+    if (context->withdrawal_credentials[0] == ETH1_ADDRESS_WITHDRAWAL_PREFIX) {
```
Comment (mkornatz) [inline src/handle_finalize.c:17]: Do we need to also check the length of `context->withdrawal_credentials` to ensure it's not just a single 0x01 byte?

Comment (mkornatz) [reply src/handle_finalize.c:17]: Seems like we do the length check in the params handler. So, nvm.

File: src/handle_query_contract_ui.c [hunk 2]
```diff
@@ -1,42 +1,64 @@
...
+static bool set_amount_ui(ethQueryContractUI_t *msg) {
+    set_screen_title(msg, "Amount");
 
     const uint8_t *eth_amount = msg->pluginSharedRO->txContent->value.value;
     uint8_t eth_amount_size = msg->pluginSharedRO->txContent->value.length;
 
     // Converts the uint256 number located in `eth_amount` to its string representation and
     // copies this to `msg->msg`.
-    amountToString(eth_amount, eth_amount_size, WEI_TO_ETHER, "ETH", msg->msg, msg->msgLength);
+    return amountToString(eth_amount,
```
Comment (mkornatz) [inline src/handle_query_contract_ui.c:24]: Why is this camelCase and other functions snake_case?

**Review by mkornatz — APPROVED** [review]
No concerns from me, but some minor comments.

File: src/handle_query_contract_ui.c (hunk 2 above)
Comment (pienkowb) [reply src/handle_query_contract_ui.c:24]: No idea 😄 This function is imported from `ethereum-plugin-sdk`: https://github.com/LedgerHQ/ethereum-plugin-sdk/blob/4d8e04480530cd14dd670a0a6405bca992d16942/include/eth_internals.h#L108-L113

File: src/handle_finalize.c (hunk 1 above)
Comment (pienkowb) [reply src/handle_finalize.c:17]: Yeah, `context->withdrawal_credentials` is always 32 bytes long and we check the length of the input parameter too ([here](https://github.com/figment-networks/ledger-app-plugin/blob/538fdfea295c60fb0e93232c8fb2f2ce59982819/src/handle_provide_parameter.c#L45-L49)).

**Review by JesseKuntz — APPROVED** [review]
🔥 LGTM! 

Tested this, was able to get the warning screen in both scenarios (when withdrawal address is different than the funding(sender/wallet) address, and when the withdrawal credentials are not the same across all validators).

IMG_6572

**Comment by jaredcrace** [comment]
My review is completed. Only 2 minor comments from me.

**Pull request merged**